packages/ui/src/modal/buy/BuyModal.tsx

```diff
@@ -93,10 +95,24 @@ export function BuyModal({
     openState ? openState[0] : false,
     openState
   )
-  const { chain: activeChain } = useNetwork()
+
+  const { chains, chain: activeChain } = useNetwork()
+  const { switchNetworkAsync } = useSwitchNetwork()
+
+  const selectedChain = chainId
```

Comment:
`modalChain` feels like a better name.

Reply:
You'll also need to loop the client chains to find the chainid passed in or the active chain (in RK, not the wallet)
You'll need to loop through the network chains to match the same chainId we found earlier and then you can get the block explorer address from that.

Reply:
This is the same logic used in the hooks:
```
  const client = useReservoirClient()
  const chain =
    chainId !== undefined
      ? client?.chains.find((chain) => chain.id === chainId)
      : client?.currentChain()
```